### PR TITLE
[TIMOB-19033] Global.String and i18n functions - READY

### PR DESCRIPTION
**Pull request by infosia**
[TIMOB-19033](https://jira.appcelerator.org/browse/TIMOB-19033) Global.String and i18n functions.

- `Ti.Locale` TitaniumKit
- Global.String TitaniumKit
- cross-platform `Ti.Locale.getString` and `L()` implementation
- cross-platform `String.format` implementation

Ready for merge, also done integration with Titanium CLI  [[TIMOB-19041](https://jira.appcelerator.org/browse/TIMOB-19041)].

## app.js
```javascript
var i18nMissingMsg = '<no translation available>';

var string1 = 'You say ' + Ti.Locale.getString('signoff', i18nMissingMsg) + ' and I say ' + Ti.Locale.getString('greeting', i18nMissingMsg) + '!';
var string2 = String.format(L('phrase'), L('greeting', i18nMissingMsg), L('signoff', i18nMissingMsg));

var win = Ti.UI.createWindow();
var label = Ti.UI.createLabel({ text: string2 });
win.add(label);

win.open();
```

## i18n/en/strings.xml

```xml
<?xml version="1.0" encoding="UTF-8"?>
<resources>
  <string name="phrase" formatted="false">You say %2$s and I say %1$s!</string>
  <string name="greeting">hello</string>
  <string name="signoff">goodbye</string>
</resources>
```

### CODE REVIEW

File: Examples/NMocha/src/Assets/ti.locale.test.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    it('Ti.Locale.getString_format', function (finish) {
+        var i18nMissingMsg = '<no translation available>';
+        var string1 = 'You say ' + Ti.Locale.getString('signoff', i18nMissingMsg) + ' and I say ' + Ti.Locale.getString('greeting', i18nMissingMsg) + '!';
+        var string2 = String.format(L('phrase'), L('greeting', i18nMissingMsg), L('signoff', i18nMissingMsg));
+
+        if (Ti.Locale.currentLanguage == 'en') {
+            should(string1).eql('You say goodbye and I say hello!');
+            should(string2).eql('You say goodbye and I say hello!');
+        } else if (Ti.Locale.currentLanguage == 'ja') {
```
Comment (sgtcoolguy) [inline Examples/NMocha/src/Assets/ti.locale.test.js]: it's too bad we have no way of "forcing" a locale to test translations and some formatting of dates/currency... We may want to have some way to do so.

File: Source/Ti/include/TitaniumWindows/Locale.hpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2015 by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License.
+ * Please see the LICENSE included with this distribution for details.
+ */
+
+#ifndef _TITANIUMWINDOWS_LOCALE_HPP_
+#define _TITANIUMWINDOWS_LOCALE_HPP_
+
+#include "Titanium/Locale.hpp"
```
Comment (sgtcoolguy) [inline Source/Ti/include/TitaniumWindows/Locale.hpp]: Is there a reason we need a Windows-specific override/subclass? It appears that there's no overridden behavior right now.

**Comment by sgtcoolguy** [comment]
@infosia We do have a section for the CLI build process intended to do the i18n resource files. Here's the relevant file for that: https://github.com/appcelerator/titanium_mobile_windows/blob/master/cli/commands/_build/generate.js

**Comment by dawsontoth** [comment]
Kota, can you rebase on master so we get the CI build fix, maybe?

**Comment by sgtcoolguy** [comment]
test this please.

**Pull request merged**